app/src/organisms/Desktop/ProtocolVisualization/StepDetailContainer.tsx

```diff
@@ -84,19 +92,25 @@ export function StepDetailContainer({
           selected={isRightPipetteActive}
         />
       ) : null}
-      <TipPickupContainer />
-      <SourceWellViewContainer protocolKey={protocolKey} />
-      <SourceLabwareContainer protocolKey={protocolKey} />
-      <DestinationWellViewContainer protocolKey={protocolKey} />
-      <DestinationLabwareContainer protocolKey={protocolKey} />
-      <TipDisposalContainer protocolKey={protocolKey} />
-      {moduleOnSlot != null ? (
-        <ModuleSlotDetails
-          moduleId={moduleOnSlot[0]}
+      {tiprackOnSlot != null && labwareEntities[tiprackOnSlot] != null ? (
+        <TipPickupContainer
+          tiprackEntity={labwareEntities[tiprackOnSlot]}
+          robotState={robotState}
+        />
+      ) : null}
+      {topMostLabwareOnSlot != null ? (
+        <LabwareSlotContainer
+          topLabwareOnSlotId={topMostLabwareOnSlot}
+          labwareEntities={labwareEntities}
+          commands={commands}
+          currentCommand={currentCommand}
+          liquids={liquids}
+          robotState={robotState}
+          pipetteEntities={pipetteEntities}
           moduleEntities={moduleEntities}
```

Comment:
looks like labwareEntities, pipetteEntities, and moduleEntities aren't being used directly within this component, so is it necessary to destructure and pass them?

```suggestion
          topLabwareOnSlotId={topMostLabwareOnSlot}
          invariantContext={invariantContext}
          commands={commands}
          currentCommand={currentCommand}
          liquids={liquids}
          robotState={robotState}

```

Reply:
it might be used later if we end up showing module state though, right? we'd need the module entities